pkg/containerwatcher/dnsmanager/dns_manager.go

```diff
@@ -0,0 +1,34 @@
+package dnsmanager
+
+import (
+	"context"
+	"node-agent/pkg/config"
+	"node-agent/pkg/k8sclient"
+	"node-agent/pkg/storage"
+
+	"github.com/goradd/maps"
+	tracerdnstype "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/dns/types"
+)
+
+// DNSManager is used to manage DNS events and save IP resolutions. It exposes an API to resolve IP address to domain name.
+type DNSManager struct {
+	addressToDomainMap maps.SafeMap[string, string] // this map is used to resolve IP address to domain name
```

Comment:
do we want to support TTL? in case DNS records change?

Reply:
I don't it is needed here. Every time I get a new query, it will override the entry on the map. The query will be performed a short time before we handle the network events, so the addresses will be always up-to-date